### PR TITLE
Feature/dummy index

### PR DESCRIPTION
**Pull request by tneuer**
### Summary

This new function allows the creation of a darts.TimeSeries from any `pd.DataFrame` or `pd.Series` if no time stamp is available. Might be useful for longitudinal data measured in steps. An artificial time stamp is added starting at 1970-01-01 00:00:00 with a frequency of 1 second.

### CODE REVIEW

**Review by hrzn — COMMENTED** [review]
For this change, I think that we should go the extra step and modify the constructor and the other methods as well, because IMO our objectives here should be that users could easily create time series with “any” DataFrame without worrying if there’s no DateTimeIndex in it.

File: darts/models/forecasting_model.py [hunk 1]
```diff
@@ -62,6 +62,13 @@ def fit(self, series: TimeSeries) -> None:
         self.training_series = series
         self._fit_called = True
 
+        if hasattr(self, "trend"):
+            raise_if(
+                self.trend and self.trend != "c" and self.training_series.has_dummy_index,
```
Comment (hrzn) [inline darts/models/forecasting_model.py:67]: I think here instead of check for `self.trend` we should probably call a function `self.supports_dummy_index()` or something like that. This way the super-class doesn't have to know about the specifics of the implementations (e.g. checking `self.trend != 'c'`. It is rather in `AutoARIMA` that we should do this check. We can have a default `ForecastingModel.supports_dummy_index()` always returning `True`, which could be overridden in `AutoARIMA`.

File: darts/tests/test_local_forecasting_models.py [hunk 2]
```diff
@@ -131,3 +132,15 @@ def test_exogenous_variables_support(self):
                 model.fit(self.ts_gaussian, exog=self.ts_gaussian[:-1])
             with self.assertRaises(ValueError):
                 model.fit(self.ts_gaussian[1:], exog=self.ts_gaussian[:-1])
+
+    def test_dummy_series(self):
```
Comment (hrzn) [inline darts/tests/test_local_forecasting_models.py:138]: 👍

File: darts/tests/test_local_forecasting_models.py [hunk 3]
```diff
@@ -131,3 +134,16 @@ def test_exogenous_variables_support(self):
...
+        values = np.random.uniform(low=-10, high=10, size=100)
+        ts = TimeSeries(pd.DataFrame({"V1": values}), dummy_index=True)
+
+        varima = VARIMA(trend="t")
+        with self.assertRaises(ValueError):
+            varima.fit(series=ts)
+
+        if PMDARIMA_AVAILABLE:
```
Comment (hrzn) [inline darts/tests/test_local_forecasting_models.py:146]: Tests were failing, I added this one.

File: darts/timeseries.py [hunk 4]
```diff
@@ -41,25 +42,38 @@ def __init__(self,
         fill_missing_dates
             Optionally, a boolean value indicating whether to fill missing dates with NaN values
             in case the frequency of `series` cannot be inferred.
+        dummy_index
+            Optionally, if no date time index is present or even available. Creates a dummy index to be
```
Comment (hrzn) [inline darts/timeseries.py:46]: I would rephrase: "Optionally, if no date time index is present, this flag will instruct Darts to build a dummy time index in order to obtain a valid TimeSeries. This option can be used in cases where the time index doesn't matter.".

File: darts/timeseries.py [hunk 5]
```diff
@@ -725,14 +740,15 @@ def from_series(pd_series: pd.Series,
         TimeSeries
             A TimeSeries constructed from the inputs.
```
Comment (hrzn) [inline darts/timeseries.py:741]: You should also add the documentation for dummy_index here.

File: darts/timeseries.py [hunk 6]
```diff
@@ -758,20 +774,21 @@ def from_dataframe(df: pd.DataFrame,
         TimeSeries
             A univariate or multivariate TimeSeries constructed from the inputs.
```
Comment (hrzn) [inline darts/timeseries.py:775]: Here too, you should document dummy_index

File: darts/timeseries.py [hunk 7]
```diff
@@ -806,6 +823,28 @@ def from_times_and_values(times: pd.DatetimeIndex,
             df.columns = columns
         return TimeSeries(df, freq, fill_missing_dates)
 
+    def create_dummy_index(self, start="19700101", freq="S") -> 'TimeSeries':
```
Comment (hrzn) [inline darts/timeseries.py:826]: Probably this method should be private (i.e. rename `_create_dummy_index`)

File: darts/timeseries.py [hunk 8]
```diff
@@ -832,7 +871,15 @@ def plot(self,
                 kwargs['figure'] = plt.gcf()
                 if 'label' in kwargs:
                     kwargs['label'] = label + '_' + str(i)
-            self.univariate_component(i).pd_series().plot(*args, **kwargs)
+            if self.has_dummy_index:
```
Comment (hrzn) [inline darts/timeseries.py:874]: 👍

File: darts/timeseries.py (hunk 7 above)
Comment (tneuer) [reply darts/timeseries.py:826]: Done

File: darts/timeseries.py (hunk 4 above)
Comment (tneuer) [reply darts/timeseries.py:46]: Done

File: darts/timeseries.py (hunk 5 above)
Comment (tneuer) [reply darts/timeseries.py:741]: Done

File: darts/timeseries.py (hunk 6 above)
Comment (tneuer) [reply darts/timeseries.py:775]: Done

**Review by hrzn — APPROVED** [review]
